File: patterns/c-patterns/pattern124.c
// Pattern 124: ...

//       **
//       **
//     *    *
//     *    *
//   *        *
//   *        *
// *            *

#include <stdio.h>

int main()
{
    int n = 7;
    int i, j;

    int x = n;
    for (i = 1; i <= n; i++)
    {
        for (j = 1; j <= 2 * n; j++)
        {
            if (j == x || j == 2 * n - x + 1)
            {
                printf("*");
            }
            else
            {
                printf(" ");
            }
        }
        printf("\n");
        if (i % 2 == 0)
            x -= 2;
    }
    return 0;
}

File: patterns/javascript-patterns/pattern124.js
// Pattern 124: ...

//       **
//       **
//     *    *
//     *    *
//   *        *
//   *        *
// *            *

const n = 7;
let x = n;

for (let i = 1; i <= n; i++) {
    let row = '';
    for (let j = 1; j <= 2 * n; j++) {
        if (j === x || j === 2 * n - x + 1) {
            row += '*';
        } else {
            row += ' ';
        }
    }
    console.log(row);
    if (i % 2 === 0) {
        x -= 2;
    }
}
